crates/sui-protocol-config/src/lib.rs

```diff
@@ -1911,7 +1911,13 @@ impl ProtocolConfig {
 
     pub fn gc_depth(&self) -> u32 {
         if cfg!(msim) {
-            // exercise a very low gc_depth
+            // Allow for overriding the gc depth for simtests.
+            if let Ok(gc_depth) = std::env::var("SIMTEST_CONSENSUS_GC_DEPTH_OVERRIDE") {
```

Comment:
Maybe we should just add a field to override gc_depth in tests, instead of setting override via env var.